pandas/core/indexes/datetimes.py

```diff
@@ -915,7 +915,7 @@ def date_range(
     tz : str or tzinfo, optional
         Time zone name for returning localized DatetimeIndex, for example
         'Asia/Hong_Kong'. By default, the resulting DatetimeIndex is
-        timezone-naive.
+        timezone-naive unless timezone-aware datetime-likes are passed.
```

Comment:
Could you add to the `Examples` section below demonstrating this?

Reply:
@mroeschke Added. Please have a look.